utils/security/user.go

```diff
@@ -0,0 +1,217 @@
+	}
+	defer res.Body.Close()
+	body, err := ioutil.ReadAll(res.Body)
+	if string(body) != "" {
+		keycloakAPIError := parseKeycloakError(string(body), res.StatusCode)
+		keycloakAPIError.Error = errOpCreate
+		kcError := errors.New(keycloakAPIError.ErrorDescription)
+		return &SecError{keycloakAPIError.Error, kcError, kcError.Error()}
+	}
+	return nil
+}
+
+// SecUserGet : Get user from Keycloak
+func SecUserGet(c *cli.Context) (*RegisteredUser, *SecError) {
+	if c.GlobalBool("insecure") {
+		http.DefaultTransport.(*http.Transport).TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
+	}
+	hostname := strings.TrimSpace(strings.ToLower(c.String("host")))
+	realm := strings.TrimSpace(c.String("realm"))
+	accesstoken := strings.TrimSpace(c.String("accesstoken"))
+	searchName := strings.TrimSpace(c.String("name"))
+
+	// authenticate if needed
+	if accesstoken == "" {
+		authToken, err := SecAuthenticate(c, KeycloakMasterRealm, KeycloakAdminClientID)
+		if err != nil || authToken == nil {
+			return nil, err
+		}
+		accesstoken = authToken.AccessToken
+	}
+
+	// build REST request
+	url := hostname + "/auth/admin/realms/" + realm + "/users?username=" + searchName
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return nil, &SecError{errOpConnection, err, err.Error()}
+	}
+	req.Header.Add("Authorization", "Bearer "+accesstoken)
+	req.Header.Add("cache-control", "no-cache")
+	req.Header.Add("cache-control", "no-cache")
+	res, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, &SecError{errOpConnection, err, err.Error()}
+	}
+
+	defer res.Body.Close()
+
+	// handle HTTP status codes
+	if res.StatusCode != 200 {
+		body, _ := ioutil.ReadAll(res.Body)
+		err = errors.New(string(body))
+		return nil, &SecError{errOpResponse, err, err.Error()}
+	}
+
+	registeredUsers := RegisteredUsers{}
+	body, err := ioutil.ReadAll(res.Body)
+	err = json.Unmarshal([]byte(body), &registeredUsers.Collection)
+	if err != nil {
+		return nil, &SecError{errOpResponseFormat, err, err.Error()}
+	}
+
+	registeredUser := RegisteredUser{}
+
+	if len(registeredUsers.Collection) > 0 {
+		registeredUser = registeredUsers.Collection[0]
+		return &registeredUser, nil
+	}
+
+	// user not found
+	errNotFound := errors.New(textUserNotFound)
+	return nil, &SecError{errOpNotFound, errNotFound, errNotFound.Error()}
+
+}
+
+// SecUserSetPW : Resets the users password in keycloak to a new one supplied
+func SecUserSetPW(c *cli.Context) *SecError {
+	if c.GlobalBool("insecure") {
+		http.DefaultTransport.(*http.Transport).TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
+	}
+	hostname := strings.TrimSpace(strings.ToLower(c.String("host")))
+	realm := strings.TrimSpace(c.String("realm"))
+	accesstoken := strings.TrimSpace(c.String("accesstoken"))
+	newPassword := strings.TrimSpace(c.String("newpw"))
+
+	// validate password characters
+	if strings.Contains(newPassword, "'") || strings.Contains(newPassword, "\"") {
```

Comment:
Is it just quotes that are banned from passwords? Is that a keycloak limitation?

Reply:
I was a bit nervous about allowing quotes in passwords to avoid any injection or invalid escape sequences being posted in JSON.

Reply:
I tried a few combinations with double and single quotes which did marshal ok,  and I was able to login with a changed password through the CLI and though a browser so I have removed this check.